### PR TITLE
Improve vouchers ui

### LINKED ISSUE #4262: UX improvements on the Vouchers page
Currently, Vouchers page in Dashboard 2.0 has few UX problems and it's difficult to understand. We've redesigned the whole section a bit to make it cleaner. New designs can be found on Figma. The task is to reimplement the components on vouchers page to match the designs.

image

### PR DESCRIPTION
**Pull request by benekex2**
I want to merge this change because... Resolves #4262 

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] Database migration files are up to date.
1. [ ] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [ ] GraphQL schema and type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

File: saleor/static/dashboard-next/discounts/components/VoucherValue/VoucherValue.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            />
+            <FormSpacer />
+            {variant === "update" ? (
+              <>
+                <RadioGroupField
+                  choices={voucherTypeChoices}
+                  disabled={disabled}
+                  error={!!errors.type}
+                  hint={errors.type}
+                  label="Discount Specific Information"
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/discounts/components/VoucherValue/VoucherValue.tsx]: Not translated

**Comment by django-queries** [comment]
Here is the report for c005c2f443d9083e90d61d7413e40065855ff524 (mirumee/saleor @ improve-vouchers-ui)
Base comparison is 43d44f54949a4d2e7ee52d901a3a25ccb35db25c.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  add billing address to checkout     	         41	         41	             24
  add shipping to checkout            	          7	          7	              0
  checkout payment charge             	         16	         16	              0
  complete checkout                   	          6	          6	              0
  create checkout                     	         45	         45	             21

# api.benchmark homepage
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve main menu                  	          5	          5	              0
  retrieve product list               	          4	          4	              0
  retrieve secondary menu             	          5	          5	              0
  retrieve shop                       	          2	          2	              0

# api.benchmark product
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  product details                     	         13	         13	              3

# api.benchmark variant
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve variant list               	          9	          9	              2
```

</p>
</details>

File: saleor/static/dashboard-next/discounts/views/VoucherCreate.tsx [hunk 2]
```diff
@@ -61,18 +52,30 @@ export const VoucherDetails: React.StatelessComponent = () => {
                 voucherCreate({
                   variables: {
                     input: {
+                      applyOncePerOrder: formData.applyOncePerOrder,
                       code: formData.code,
-                      discountValue: decimal(formData.value),
-                      discountValueType: discountValueTypeEnum(
-                        formData.discountType
+                      discountValue:
+                        formData.discountType.toString() === "SHIPPING"
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/discounts/views/VoucherCreate.tsx]: Compare to `enum` values

File: saleor/static/dashboard-next/discounts/views/VoucherDetails.tsx [hunk 3]
```diff
@@ -319,29 +310,42 @@ export const VoucherDetails: React.StatelessComponent<VoucherDetailsProps> = ({
                               activeTab={params.activeTab}
                               onBack={() => navigate(voucherListUrl())}
                               onTabClick={changeTab}
-                              onSubmit={formData =>
+                              onSubmit={formData => {
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/discounts/views/VoucherDetails.tsx]: Unnecesary brace

File: saleor/static/dashboard-next/misc.ts [hunk 4]
```diff
@@ -236,3 +236,28 @@ export function stopPropagation(cb: () => void) {
     cb();
   };
 }
+
+export function connectDateTime(date: string, time?: string) {
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/misc.ts]: How about `joinDateTime`?

File: saleor/static/dashboard-next/misc.ts [hunk 5]
```diff
@@ -236,3 +236,28 @@ export function stopPropagation(cb: () => void) {
...
+
+export function splitDateTime(dateTime: string) {
+  if (!dateTime) {
+    return {
+      date: "",
+      time: ""
+    };
+  }
+  const splitDateTime = moment(dateTime)
+    .format("YYYY-MM-DD HH:mm")
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/misc.ts]: Can we add comment here that this is html input tags formatting?

File: saleor/static/dashboard-next/discounts/components/VoucherCreatePage/VoucherCreatePage.tsx [hunk 6]
```diff
@@ -72,18 +82,50 @@ const VoucherCreatePage: React.StatelessComponent<VoucherCreatePageProps> = ({
...
                 variant="create"
+              />
+              <CardSpacer />
+              <VoucherTypes
+                data={data}
+                disabled={disabled}
+                errors={formErrors}
                 onChange={change}
               />
+              {data.discountType.toString() !== "SHIPPING" ? (
```
[the remark this answers is not in the document]
Comment (benekex2) [reply saleor/static/dashboard-next/discounts/components/VoucherCreatePage/VoucherCreatePage.tsx]: `DiscountValueTypeEnum.SHIPPING` not exist. This is only visual radio for user. Thats why I do not use `enum`

File: saleor/static/dashboard-next/discounts/components/VoucherInfo/VoucherInfo.tsx [hunk 7]
```diff
@@ -1,101 +1,64 @@
...
-    }
-  });
+const VoucherInfo = ({
+  data,
+  disabled,
+  errors,
+  variant,
+  onChange
+}: VoucherInfoProps) => {
+  const [generatedCode, setGeneratedCode] = React.useState();
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/discounts/components/VoucherInfo/VoucherInfo.tsx]: Still can't understand why you need a state for this

File: saleor/static/dashboard-next/discounts/components/VoucherLimits/VoucherLimits.tsx [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+  errors: FormErrors<"usageLimit">;
+  onChange: (event: React.ChangeEvent<any>) => void;
+}
+
+const styles = (theme: Theme) =>
+  createStyles({
+    root: {
+      display: "grid",
+      gridColumnGap: theme.spacing.unit * 2 + "px",
+      gridTemplateColumns: "1fr"
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/discounts/components/VoucherLimits/VoucherLimits.tsx]: This isn't even a grid

File: saleor/static/dashboard-next/discounts/components/VoucherValue/VoucherValue.tsx [hunk 9]
```diff
@@ -0,0 +1,112 @@
+import Card from "@material-ui/core/Card";
+import CardContent from "@material-ui/core/CardContent";
+import Typography from "@material-ui/core/Typography";
+import * as React from "react";
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/discounts/components/VoucherValue/VoucherValue.tsx]: Asterisk import

File: saleor/static/dashboard-next/discounts/components/VoucherInfo/VoucherInfo.tsx [hunk 10]
```diff
@@ -1,101 +1,64 @@
...
-    }
-  });
+const VoucherInfo = ({
+  data,
+  disabled,
+  errors,
+  variant,
+  onChange
+}: VoucherInfoProps) => {
+  const onGenerateCode = () => {
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/discounts/components/VoucherInfo/VoucherInfo.tsx]: You can use shorthand 
```jsx
const onGenerateCode = () => onChange(...)
```

File: saleor/static/dashboard-next/discounts/components/VoucherInfo/VoucherInfo.tsx [hunk 11]
```diff
@@ -1,101 +1,64 @@
...
-      </Card>
-    );
-  }
-);
+  return (
+    <Card>
+      <CardTitle
+        title={i18n.t("General Information")}
+        toolbar={
+          variant === "create" ? (
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/discounts/components/VoucherInfo/VoucherInfo.tsx]: `&&` would be more appropriate here

File: saleor/static/dashboard-next/discounts/components/VoucherValue/VoucherValue.tsx [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+          label={i18n.t("Discount Value")}
+          value={data.value}
+          type="number"
+          fullWidth
+          inputProps={{
+            min: 0
+          }}
+        />
+        <FormSpacer />
+        {variant === "update" ? (
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/discounts/components/VoucherValue/VoucherValue.tsx]: `&&` would be better choice

**Pull request merged**